src/python_testing/TC_ACL_2_6.py

```diff
@@ -136,26 +154,69 @@ async def test_TC_ACL_2_6(self):
         asserts.assert_equal(result[0].Status, Status.Success, "Write should have succeeded")
 
         self.step(5)
-        # Create correct event path with endpoint 0
-        events_response2 = await self.default_controller.ReadEvent(
-            nodeid=self.dut_node_id,
-            events=event_path,
-            fabricFiltered=True,
-            eventNumberFilter=max(initial_event_num)+1
+        # Wait for subscription events
+        received_subscription_events = []
+        for _ in range(2):
+            event_data = events_callback.wait_for_event_report(acec_event, timeout_sec=15)
+            logging.info(f"Received subscription event: {event_data}")
+            received_subscription_events.append(event_data)
+
+        # Read events with retries
+        events_response = await self.th1.ReadEvent(
+            self.dut_node_id,
+            events=[(0, acec_event)],
+            fabricFiltered=True
         )
+        logging.info(f"Read events response: {events_response}")
+
+        if events_response:
+            # Get the most recent events from the read response
+            read_events = sorted([e.Data for e in events_response],
+                                 key=lambda x: next(e.Header.EventNumber for e in events_response if e.Data == x),
+                                 reverse=True)[:2]
+
+        # Verify we got the expected number of events
+        asserts.assert_true(len(read_events) >= 2,
```

Comment:
I think this might end up erroring weirdly if you don't have events_response.
```suggestion
        asserts.assert_true(events_response, "Did not receive a response when calling ReadEvents")
        asserts.assert_true(len(read_events) >= 2,
```

Reply:
uh...no guarantees as to the correctness of that code - I wrote it in a comment, bad things may happen.

Reply:
Thank you, the new assert you added appears works perfectly, validated in local setup before adding to the code.